api/middleware.go

```diff
@@ -27,3 +29,50 @@ func addGetBody(w http.ResponseWriter, req *http.Request) (context.Context, erro
 	req.Body, _ = req.GetBody()
 	return req.Context(), nil
 }
+
+func (api *API) loadInstanceConfig(w http.ResponseWriter, r *http.Request) (context.Context, error) {
+	ctx := r.Context()
+
+	instanceID := r.Header.Get("x-nf-id")
+	if instanceID == "" {
+		return nil, badRequestError("Netlify microservice headers missing")
+	}
+
+	// TODO verify JWS of microservice API request
+
+	logEntrySetField(r, "instance_id", instanceID)
+	instance, err := api.db.GetInstance(instanceID)
+	if err != nil {
+		if models.IsNotFoundError(err) {
+			return nil, notFoundError("Unable to locate site configuration")
+		}
+		return nil, internalServerError("Database error loading instance").WithInternalError(err)
+	}
+
+	env := r.Header.Get("x-nf-env")
```

Comment:
Definitely. I was waiting for a spec I could implement against before starting down that path.